### PR TITLE
Adds support for AND, OR, custom keyword matching

### PR DESCRIPTION
**Pull request by JPrevost**
NOTE: while you can use the GraphQL playground to review these changes, there is also an experimental branch of TIMDEX-UI deployed and configured to use this branch that may also be helpful to more easily see the effects of these proposed changes.

Why are these changes being introduced:

* TIMDEX currently ORs all keyword searches together which leads to a lot of search results, many of which are not relevant (to the extreme of sometimes having no records returned that have all the terms which is sometimes the expectation)

Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/GDT-281

How does this address that need:

* Leverages OpenSearch `minimum_should_match` [feature](https://opensearch.org/docs/latest/query-dsl/minimum-should-match/#valid-values) along with a GraphQL feature to choose a preconfigured value or supply your own

Document any side effects to this change:

* once we get feedback from stakeholders, it is likely we'll want to adjust the options available, document them more thoroughly in the API docs, and provide them with meaningful names, as well as probably remove some of the options

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/graphql/types/query_type.rb
Comment (JPrevost) [inline app/graphql/types/query_type.rb:63]: Yeah so these docs are already out of date with the code. Will fix before merge.

File: app/models/opensearch.rb
Comment (JPrevost) [inline app/models/opensearch.rb:111]: I'm open to holding off on merging until we decide what we want here fwiw. I just needed something in place for stakeholder feedback.

**Review by jazairi — COMMENTED** [review]
I think this all works as expected, insofar as I trust myself to validate code that involves numbers. I left a comment about the documentation of what I assume became `experiment_a` in the `boolean_type` argument. I'd be fine either with documenting all possible options or just AND and OR (until we decide which experiments will become permanent).

It feels like it would be nice to provide useful information if a bad `boolean_type` were entered. I'm not sure how to achieve that in GraphQL, though, and I don't think that's something we do with other fields.

File: app/graphql/types/query_type.rb
Comment (jazairi) [reply app/graphql/types/query_type.rb]: Ha, you beat me to it. Ignore my comment below.

File: app/models/opensearch.rb
Comment (jazairi) [reply app/models/opensearch.rb:111]: That's an interesting idea. As noted below, I have some concerns about introducing/documenting experimental features in the API, since we may remove one or more of them. It's a fairly mild concern, though, since I don't think anyone but us is going to be using these before we decide what we want to do with them.

**Review by jazairi — APPROVED** [review]

File: app/models/opensearch.rb
Comment (JPrevost) [reply app/models/opensearch.rb:111]: I opted for just documented AND and OR and merging as that'll leave the timdex-ui experimental PR build working for stakeholders (the API still shuts down PRs after a day). I apologize to our future selves when we forget to cleanup these experiments and can't remember why they are there and aren't documented, etc.

**Pull request merged**
